src/psyclone/tests/psyad/transformations/test_preprocess.py

```diff
@@ -64,6 +64,30 @@ def test_preprocess_no_change():
     assert result == code
 
 
+def test_preprocess_reference(tmpdir, fortran_reader, fortran_writer):
+    '''Test the preprocess script replaces array references with
+    references to arrays.
+
+    '''
+    code = (
+        "program test\n"
+        "real :: a(10), b(10), c(10)\n"
+        "a = b * c\n"
+        "end program test\n")
+    expected = (
+        "program test\n"
+        "  real, dimension(10) :: a\n"
+        "  real, dimension(10) :: b\n"
+        "  real, dimension(10) :: c\n\n"
+        "  a(:) = b(:) * c(:)\n\n"
```

Comment:
How come we don't end up with an explicit loop here?

Reply:
This test has been replaced with one that shows an explicit loop being generated.